### ISSUE TITLE
Buffering video unrelated to connection

### ISSUE THREAD

**Issue by n0rder** [TeamNewPipe/NewPipe#6510]
[Screen recording](https://linksharing.samsungcloud.com/5PFO7lkTlSVj) 

As you can see in the screen record videos randomly buffer when in video mode, however in background mode the audio plays fine.

This issue is limited to newpipe and other video apps work fine.

Samsung galaxy s10+ android 11

Reinstalling newpipe doesn't work
Changing Internet connection doesn't work

**Comment by matheusAMDS** [comment]
I can't visualize the link you shared, but i'm having problems with my videos infinitely loading. Maybe its something related?

**Comment by LauriKat** [comment]
I am having same issue as yoy.

Redmi Note 8 pro
Android 10

**Comment by botistreamer** [comment]
I am also having same issue in 0.21.5 on some videos randomly. 
Poco X3 - Android 11

**Comment by B0pol** [comment]
Same

**Comment by djuarezr** [comment]
Same here, even though with version 0.21.5 the problem has been mitigated a lot. If logs are needed just let me know.

Xiaomi Mi 9T - Miui 12.0.5 - Android 10

**Comment by Redirion** [comment]
please try this apk: https://github.com/TeamNewPipe/NewPipeExtractor/pull/604

if the problem also happens with that apk, please collect the adb logcat logs from the issue

**Comment by mhmdanas** [comment]
For all those facing this issue, using external players will probably not have this issue.

**Comment by djuarezr** [comment]
It is the first time I try to get a logcat so I do not know whether it is what you need, but I got it. The logcat is from TeamNewPipe/NewPipeExtractor#604. Video starts at line 12, and after 30 seconds it starts the continuous buffering.

Hope this helps.

[NewPipe_YoutubeI_Api.log](https://github.com/TeamNewPipe/NewPipe/files/6676465/NewPipe_YoutubeI_Api.log)

**Comment by Redirion** [comment]
thanks. A quick glance shows these log lines demand further inspection as they might be the cause (but not the root cause) of the issue:

AudioTrack: getTimestamp_l(417): retrograde timestamp time corrected
AudioTrack: getTimestamp_l(417): device stall time corrected using current time

**Comment by Redirion** [comment]
@djuarezr could you try if the issue also happens with Media Tunneling disabled? You find the option to disable that in the debug options.

**Comment by djuarezr** [comment]
First I should have mentioned before that the log that I posted before is from a video at 1.5x speed. So far it also happens when tunneling is disabled. But I could not replicate the issue with the same video anymore. Thus, to be more precise, here is a more fair comparison I made. Every log is from a clean app, 1080p webM video, and the smartphone at the exact same distance to WiFi:

- Video at 1x: just one buffering at the beginning.
[NewPipe_YoutubeI_Api_1x.log](https://github.com/TeamNewPipe/NewPipe/files/6678771/NewPipe_YoutubeI_Api_1x.log)

- Video at 1x Tunneling disabled: just one buffering at the beginning. 
[NewPipe_YoutubeI_Api_Media_Tunneling_Disabled_1x.log](https://github.com/TeamNewPipe/NewPipe/files/6678772/NewPipe_YoutubeI_Api_Media_Tunneling_Disabled_1x.log)

- Video at 1.5x: just one buffering at the beginning.
[NewPipe_YoutubeI_Api_1.5x.log](https://github.com/TeamNewPipe/NewPipe/files/6678828/NewPipe_YoutubeI_Api_1.5x.log)

- Video at 1.5x Tunneling disabled: More than one buffering issue while playing. 
[NewPipe_YoutubeI_Api_Media_Tunneling_Disabled_1.5x.log](https://github.com/TeamNewPipe/NewPipe/files/6678835/NewPipe_YoutubeI_Api_Media_Tunneling_Disabled_1.5x.log)

Hope all these logs could help to decrypt the issue.

**Comment by lordkitsuna** [comment]
ill throw my hat in. i am having issues with newpipe buffering super slowly initially on a 1Gbps connection (530Mbps verified as possible on wifi locally with iperf3)  video at any quality stalls for upwards of 30 seconds, after that they play fine unless i seek out far then it happens again. fresh install of lineageos, tons of free space, no gapps or anything so cpu is pretty quiet.

happy to provide any info i can just let me know what you need

**Comment by Taratect** [comment]
I used to have this problem till the version 0.21.5 came out. So far there hasn't been any buffering.
Hope if updating helps n I'll update if I face buffering while using the latest version atm.

Also I'd like to let I know as @djuarezr pointed, I also watch my vids by increasing playback speed of this is somehow related.

**Comment by stoneubi** [comment]
Same here. Since two or three versions this buffering problem occurs. Before that, everything worked perfectly. Latest version hasn't fixed this. Realme 6. Hope this issue will be fixed as for now newpipe cannot play every video.

**Comment by nikhilCad** [comment]
I do not know whether it is related to this issue or not but loading large videos like 10hr+ take a very long time as compared to the normal 10-15 minute videos.

**Comment by vahe554-lab** [comment]
exactly, same problem 2 smartphones, redmi 8note, redmi 9t, version 0.21.5

**Comment by dankxylese** [comment]
I'll add some of my experience with this bug. 

Killing the app and restarting it fixes the problem for me (I have long press back bind to kill apps).

It will usually hover over 250-280KB/s for me when it's stuck in that bugged state. After killing and restarting the app, going to history and playing the same video, it instantly peeks to 4-6MB/s and nicely loads the video.

My guess would be that it might be YouTube figuring out the client is "not official" and throttling the connection. Or maybe there are slow servers that NewPipe is connecting to, since just killing the app, no cache clear or anything fixes the issue for me.


Android 10 AICP (AOSP based).
Rooted + Magisk and Xposed modules loaded.
OnePlus 5T.


I'll make sure to grab a logcat when it happens next time for me, so I can try and capture the before and after killing the app. Maybe it'll help lead to the issue.

**Comment by dankxylese** [comment]
> First I should have mentioned before that the log that I posted before is from a video at 1.5x speed. So far it also happens when tunneling is disabled. But I could not replicate the issue with the same video anymore. Thus, to be more precise, here is a more fair comparison I made. Every log is from a clean app, 1080p webM video, and the smartphone at the exact same distance to WiFi:
> 
> * Video at 1x: just one buffering at the beginning.
>   [NewPipe_YoutubeI_Api_1x.log](https://github.com/TeamNewPipe/NewPipe/files/6678771/NewPipe_YoutubeI_Api_1x.log)
> * Video at 1x Tunneling disabled: just one buffering at the beginning.
>   [NewPipe_YoutubeI_Api_Media_Tunneling_Disabled_1x.log](https://github.com/TeamNewPipe/NewPipe/files/6678772/NewPipe_YoutubeI_Api_Media_Tunneling_Disabled_1x.log)
> * Video at 1.5x: just one buffering at the beginning.
>   [NewPipe_YoutubeI_Api_1.5x.log](https://github.com/TeamNewPipe/NewPipe/files/6678828/NewPipe_YoutubeI_Api_1.5x.log)
> * Video at 1.5x Tunneling disabled: More than one buffering issue while playing.
>   [NewPipe_YoutubeI_Api_Media_Tunneling_Disabled_1.5x.log](https://github.com/TeamNewPipe/NewPipe/files/6678835/NewPipe_YoutubeI_Api_Media_Tunneling_Disabled_1.5x.log)
> 
> Hope all these logs could help to decrypt the issue.

I haven't messed with playback speed, so probably isn't related to it since I'm getting it too. In my case it's one to 3-4 times a day, usually when Newpipe has been left in sleep for a while, or is a cold start (OnStart()/OnResume())? 

Or could be YouTube breaking things again. Building an apk with the old exoplayer should rule that one out.

**Comment by mhmdanas** [comment]
> Or could be YouTube breaking things again. Building an apk with the old exoplayer should rule that one out.

That's precisely what #6524 is.

**Comment by bzp99** [comment]
Same issue here, although I haven’t tried with v0.21.5 yet. Even on extremely low quality (say 360p), videos keep buffering every few seconds, making them impossible to watch. The network speed is fine, on another phone with the official app, there is no problem. Soft restarting or killing the app does not help. I am not sure, but I think this only happens with some videos.

**Comment by opusforlife2** [comment]
Everyone facing the buffering issue, please test the APK in #6524. We wish to confirm that it fixes the problem.

**Comment by djuarezr** [comment]
I tried the APK in #6524 (could not get logs yet) and looks like the bug is gone. I had to play videos at greater than 2x speed to force some buffering. But I want other people to try the APK to compare.

Btw, I configured the APK to play videos at 1080p60 by default. Some videos I tried are not available at 1080p in youtube (just 720p or below), but newpipe allows me to play that video at 1080p60. Is there any rendering to upscale the resolution that could lead to this buffering issue?

**Comment by IHateTheProgrammingClass** [comment]
Similar issue. Sometimes the bandwidth stays around 50-100KB/s 1080p and the video buffers, plays 0.5 seconds, then repeat. Cache wipe usually solves issue.

**Comment by SeongGino** [comment]
The #6542 artifact definitely seems to have alleviated the issue on my device that has been suspect to the hitching buffering on the last couple of versions.

**Comment by Donkey-Doug** [comment]
Experiencing the same issues. Force closing the newpipe app and opening the same video again makes the video play normally. (Sometimes it is required to force close the app multiple times because it is not always fixed after 1 force close).

**Comment by SeongGino** [comment]
The issue only seems to have been diminished somewhat, but by no means 'fixed' by the test build. Looks to be the longer it's used from a fresh install, the less likely it's able to run ahead, and the more likely intermittent buffering happens.

**Comment by litetex** [comment]
Issue persists on my device even with #6542 
However on most occasions deleting the cache and restarting the app helps.

**Comment by litetex** [comment]
Has someone tried to [debug the exoplayer](https://exoplayer.dev/debug-logging.html)?

I added ``simpleExoPlayer.addAnalyticsListener(new EventLogger(trackSelector));`` after https://github.com/TeamNewPipe/NewPipe/blob/484c852efd55eb2d0279c01d19529ea5b9d59f0a/app/src/main/java/org/schabi/newpipe/player/Player.java#L478 but so far I haven't had any buffering and didn't see any errors in the log.

**Comment by djuarezr** [comment]
I (finally) got a log with the buffering issue at 1x (1080p60 webM in audio and video). The apk I used is the one from #6542. Hope this can help.

[ExoDowngrade2.log](https://github.com/TeamNewPipe/NewPipe/files/6707316/ExoDowngrade2.log)

**Comment by TiA4f8R** [comment]
The real issue is a YouTube change: see ytdl-org/youtube-dl#29326 and [this comment for a solution](https://github.com/ytdl-org/youtube-dl/issues/29326#issuecomment-865985377). It requires an extractor change.

**Comment by opusforlife2** [comment]
Ah. Makes sense. @Redirion As other users have reported, even with downgraded Exoplayer the buffering issue occurs sometimes, though it's much rarer (for me at least).

**Comment by litetex** [comment]
Could also catch the problem while sitting on the coach, attached my phone to the pc and got the logs:
Getting every few hundred ms the following message:
```
2021-06-24 21:55:43.966 8646-8646/org.schabi.newpipe.debug.exp E/EventLogger: audioTrackUnderrun [eventTime=894.44, mediaPos=315.91, window=0, period=0, 44100, 250, 4597]
```

When tapping the play-button also this happens:
```
2021-06-24 21:55:30.231 8646-11618/org.schabi.newpipe.debug.exp E/AudioTrack: StreamType not music do not upload bigdata
```

More here: [snippet.txt](https://github.com/TeamNewPipe/NewPipe/files/6711867/loggingSnippet.txt)

**Comment by NSTAdventure** [comment]
For developer consider use exoplayer v1 as videi player for now 

not all devices able to handle exoplayer v2  this one of cause why user on YouTube experience lagging blah blah blah

if newpipe use v2 consider it to downgrade v1

**Comment by Redirion** [comment]
> if newpipe use v2 consider it to downgrade v1

sorry but that's nonsense

**Comment by Dsw77** [comment]
> [Screen recording](https://linksharing.samsungcloud.com/5PFO7lkTlSVj)
> 
> As you can see in the screen record videos randomly buffer when in video mode, however in background mode the audio plays fine.
> 
> This issue is limited to newpipe and other video apps work fine.
> 
> Samsung galaxy s10+ android 11
> 
> Reinstalling newpipe doesn't work
> Changing Internet connection doesn't work

For me the buffering also happens with only background audio, randomly. Today it seemed to be resolved bij connecting over vpn,  though on the same network.

Dennis

**Comment by TacoTheDank** [comment]
> if newpipe use v2 consider it to downgrade v1

![image](https://user-images.githubusercontent.com/32376686/123472126-bf9d5e80-d5c4-11eb-9688-78b448e68203.png)

**Comment by conkerts** [comment]
I have also issues the last 2 weeks or so. I can't tell the precise date when it started.
But what I observed was, that videos will stutter, but when I start playing them in background, for whatever reason they (/audio) work then.

Another weird thing is that Android reported network activity around only  50 or 60KB/s, which is obviously not enough. 
I encountered that low speed on yt-dl, too.
And yesterday I had stuttering even in firefox on my Laptop.

I suspect YT changed something on their side, too

**Comment by NSTAdventure** [comment]
> I suspect YT changed something on their side, too

yt use exoplayer v2 that not all devices can handle that unfortunately

**Comment by NSTAdventure** [comment]
> > if newpipe use v2 consider it to downgrade v1
> 
> sorry but that's nonsense

hey it's real believe i also experienced same issuee

**Comment by Redirion** [comment]
what is your device model @NSTAdventure ?

**Comment by NSTAdventure** [comment]
> what is your device model @NSTAdventure ?

mine is realne 5i 🙏

again as i said not all devices can handle v2 if yoy doesn't want to downgrade at least add option to revert exoplayer to v1 like yt vance version 15.43.32

**Comment by litetex** [comment]
@NSTAdventure

Some comments to you statements

> not all devices able to handle exoplayer v2 this one of cause why user on YouTube experience lagging blah blah blah

→ https://exoplayer.dev/supported-devices.html 
Exoplayer supports most use cases as far back as Android 4.1; That covers 99,8% of all running android devices

> For developer consider use exoplayer v1 as videi player for now
...
> if newpipe use v2 consider it to downgrade v1
...
> yt use exoplayer v2 that not all devices can handle that unfortunately

Downgrading Exoplayer has nothing to do with this issue as stated above in https://github.com/TeamNewPipe/NewPipe/issues/6510#issuecomment-867789285. That's a completely different issue.
Btw: Exoplayer 1 was last released in  2017.

> since yt v16.14.34 they use exoplayer v2
> i already updated yt but well it's lagging like h*ll
> i tried to force codex VP9 actually it's worked but my battery are draining more faster and faster

I presume you are talking about the YT app (or YT vanced)? 
NewPipe has no option to force VP9 as far as I know.
Anyhow, this has nothing to do with NewPipe itself. 
You didn't state somewhere that you are even using the NewPipe app.
→ However everything is completely off-topic; Maybe the comments should be marked as that (@TeamNewPipe) so others - that actually want to contribute something useful - don't get confused.

> hey it's real believe me i also experienced same issues
...
> not all devices able to handle exoplayer v2 this one of cause why user on YouTube experience lagging blah blah blah
...
> come on bro I don't want force my devices due to battery hogging :(

It would also be great if you could express yourself in a nicer, more upscale way, that actually contributes something to this issue.

**Comment by NSTAdventure** [comment]
since what version of newpipe do you use exoplayer version 2?I want to download the latest version of newpipe which is still using exoplayer version 1 if it is available thanks 😊

**Comment by litetex** [comment]
@NSTAdventure 
I took a look into the commits and NewPipe was never shipped with ExoPlayer 1.

ExoPlayer was implemented in 2017 (→ https://github.com/TeamNewPipe/NewPipe/pull/486/files#diff-51a0b488f963eb0be6c6599bf5df497313877cf5bdff3950807373912ac1cdc9R48 ) and there it was already version 2.
So as far as I know there is or ever existed no version with ExoPlayer 1.

If you have further questions, I think it would be great if you could ask these in the chat: https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md#communication
This issue is definitely the wrong place for that.

**Comment by NSTAdventure** [comment]
> @NSTAdventure
> I took a look into the commits and NewPipe was never shipped with ExoPlayer 1.
> 
> ExoPlayer was implemented in 2017 (→ https://github.com/TeamNewPipe/NewPipe/pull/486/files#diff-51a0b488f963eb0be6c6599bf5df497313877cf5bdff3950807373912ac1cdc9R48 ) and there it was already version 2.
> So as far as I know there is or ever existed no version with ExoPlayer 1.
> 
> If you have further questions, I think it would be great if you could ask these in the chat: https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md#communication
> This issue is definitely the wrong place for that.

thanks i will create issues later if i face lagging too on newpipe :) 🙏

**Comment by NoblePink** [comment]
> Another weird thing is that Android reported network activity around only 50 or 60KB/s, which is obviously not enough.
> I encountered that low speed on yt-dl, too.
> And yesterday I had stuttering even in firefox on my Laptop.
> 
> I suspect YT changed something on their side, too

I agree with @conkerts on the last line. When I tried to play videos externally with MPV I got extremely low speeds. Then I killed the Newpipe app and try again on the same video it was smooth again.

**Comment by conkerts** [comment]
> I agree with @conkerts on the last line. When I tried to play videos externally with MPV I got extremely low speeds. Then I killed the Newpipe app and try again on the same video it was smooth again.

I wanted to add some "research" that I just did today, as like I said, I encountered that low speed on yt-dl, too. And the speed is just suspiciously low, which I suspected is throttling.

Well, turns out, the yt-dl guys have their own slowdown issue here:
https://github.com/ytdl-org/youtube-dl/issues/29326

Esp check out this post from @awojnowski 
https://github.com/ytdl-org/youtube-dl/issues/29326#issuecomment-865985377 : 

> I have the solution for this issue. I do not have the bandwidth to actually implement it in the source, but this should be more than enough information to do so.
> 
> The issue is that YouTube is modifying the `n` query parameter on the video playback URLs in a very similar fashion as the signature cipher. There's a pure function in the player JavaScript which takes the `n` parameter as input and outputs an `n` parameter which is not subject to throttling.
> 
> As an example, let's look at https://www.youtube.com/s/player/52dacbe2/player_ias.vflset/et_EE/base.js. The code in question which modifies `n` is as follows:
> 
> ```
> a.C&&(b=a.get("n"))&&(b=Dea(b),a.set("n",b))}};
> ```
> 
> In this case, `Dea` is the function we are looking for: [...]
> This does change with different player versions, so youtube-dl will need to extract this for every video that it fetches and then modify the n parameter as such.
> Hope this is helpful.

So the cat and mice game continues ...

**Comment by Taratect** [comment]
I'd like to mention that restarting the device puts a band aid to the problem until it recurs again.

Also, it happens when I watch vids in 2x timespeed.

**Comment by dankxylese** [comment]
> I'd like to mention that restarting the device puts a band aid to the problem until it recurs again.

Killing the app is enough for me. You can force close it in app details or if you're on a custom rom bind "long press back" to kill apps to make this quicker.

I think why it works is that by restarting you're forcing a new connection to be made. Whether it is a server that's not limiting the stream, or NewPipe being able to decode the stream properly, resetting the connection works for me.

**Comment by Kusurisan** [comment]
if i restart NewPipe it usually works for a few minutes, but then it re-appears again and again. it's really frustrating. i always watch youtube videos in bed before sleeping with NewPipe and because of this issue it is a nightmare. i'm now at a point where i download my "daily videos" i want to watch in bed to my phone so i don't need to stream them since its otherwise too annoying. it's so frustrating that youtube always needs to f*** around to try to prevent or annoy people. using yt-dl i had the same issues when i wanted to download videos.. it sucks so hard.

**Comment by TiA4f8R** [comment]
Can you try the APK in TiA4f8R#1? There should be no buffering issues with it. Use it as a temporary fix for now.

**Comment by spvkgn** [comment]
> Can you try the APK in [TiA4f8R#1](https://github.com/TiA4f8R/NewPipe/pull/1)? There should be no buffering issues with it. Use it as a temporary fix for now.

This works fine for me, there is no issue with buffering.

**Comment by SeongGino** [comment]
> 
> 
> Can you try the APK in [TiA4f8R#1](https://github.com/TiA4f8R/NewPipe/pull/1)? There should be no buffering issues with it. Use it as a temporary fix for now.

Seems to be just as bad as before, unfortunately.

**Comment by bibin-v2020** [comment]
> Can you try the APK in [TiA4f8R#1](https://github.com/TiA4f8R/NewPipe/pull/1)? There should be no buffering issues with it. Use it as a temporary fix for now.

Seems the issue fixed on this apk. I was testing it for last two days. No buffering issue reported till now

**Comment by TiA4f8R** [comment]
@SeongGino Can you share the **streaming** URL and also the YouTube URL of a video on which you have the buffering issue? I want to check something.

In order to do this, install Intent Intercept from F-Droid, enable the `Use external video player` option in Settings -> Video and audio, choose Intent Intercept in the `Open with chooser` from the Android system and post the URL you get (which contains `.googlevideo.com`).
**Don't forget to remove your IP adress in the IP field** by changing it to `XX.XX.XX` (in order to get `&ip=XX.XX.XX`). Thank you in advance.

**Comment by int3rw3bs** [comment]
The player seems to be buffering in a normal playing mode. i dont think its a connection issue because if i press the background button, the video plays immediately. This seems to be random, however i have noticed that it will mostly occur if you play a video without closing the app properly previously, and also only after you play a couple of videos concurrently.

**Comment by bibin-v2020** [comment]
> The player seems to be buffering in a normal playing mode. i dont think its a connection issue because if i press the background button, the video plays immediately. This seems to be random, however i have noticed that it will mostly occur if you play a video without closing the app properly previously, and also only after you play a couple of videos concurrently.

Same!!
Buffering issue is fixed but switching between different video sometimes end up with 'No response'

**Comment by XiangRongLin** [comment]
Can you guys test out this apk.  
It is a bare bones attempt to go with the solution from the youtube-dl thread that @conkerts linked here https://github.com/TeamNewPipe/NewPipe/issues/6510#issuecomment-871468397  
I'll be using this version too for a few days. If it works, then I'll make a proper PR out of it.

Only videos are adjusted, audio streams are not adjusted

[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/6793074/app-debug.zip)

Changes are:
https://github.com/XiangRongLin/NewPipeExtractor/commit/02d6edd9dc762cae68cd24c026576c499ce2c8dc

**Comment by LavenderSolus** [comment]
Same here. Even at 360p, it just keeps loading/buffering every two seconds. The most annoying thing about NewPipe ever.
Other apps work totally fine. Please fix it before anything else.
I use version 0.21.5

**Comment by djuarezr** [comment]
Buffering is still present with the debug app unfortunately. I noticed that the buffering bar in NewPipe is never filled as much as in other apps (or compared to youtube web itself). Could it be related with the bufferin bug?

**Comment by Thewisem** [comment]
Btw they have made 0.21.6 and told us to find issues with it. So people go check it out and see if ou have buffer issues

**Comment by Thewisem** [comment]
https://github.com/TeamNewPipe/NewPipe/issues/6577

**Comment by ShrimpBFM** [comment]
Same issue with  0.21.5 on the Google Pixel 3a running Android 11.

**Comment by Sinono3** [comment]
Same issue with 0.21.4 on Poco X3 NFC running LOS. It starts working again if I change videos. Then it stops working again. It seems it happens every 2 videos.

**Comment by SeongGino** [comment]
> 
> 
> @SeongGino Can you share the **streaming** URL and also the YouTube URL of a video on which you have the buffering issue? I want to check something.

The problem is, it's not just _a_ video that has the issue. It's _**any**_ video after the cache builds up over the course of a small handful of video streams. So I don't see how this would benefit, when/if you yourself could simply check any random video after using a specific build for a period.

**Comment by Thewisem** [comment]
Hmmm how about y'all check out the beta 0.21.6 which is out in an issue page and comment whether this buffer issue occurs in it. So far I couldn't find it

**Comment by opusforlife2** [comment]
> when/if you yourself could simply check any random video after using any build for a period.

@SeongGino The point is that none of us _can_. Our APKs from https://github.com/TiA4f8R/NewPipe/pull/1 are working just fine. If you are the only one encountering the issue still, then you are the only one who can give us the logs to investigate it.

**Comment by absNoobalo** [comment]
Same isues here @ Pixel 4a Android 11

**Comment by Flam3z** [comment]
Have the same issues semi-frequently

Samsung S10
0.21.5

**Comment by SeongGino** [comment]
> 
> 
> > when/if you yourself could simply check any random video after using any build for a period.
> 
> @SeongGino The point is that none of us _can_. Our APKs from [TiA4f8R#1](https://github.com/TiA4f8R/NewPipe/pull/1) are working just fine. If you are the only one encountering the issue still, then you are the only one who can give us the logs to investigate it.

Out of a sample size of two people (me and @spvkgn) who've commented? No offense, but that seems inconclusive - especially for a problem that accumulates over a period of time, as I've already noted in the response you quoted.

You've also said nothing of _logs_ - just URLs; that, going off of my earlier responses, could be gathered from any front page video without my intervention. Unless there's some clarification in gathering additional data unique to my situation (I don't see how that can be the case?), I've been in no position to be going through such longwinded procedures on the device I've been using with NewPipe atm just for variables that anyone else can find in my place - because the problem happens regardless of the source video being played.

Maybe in a day or so when I have the time and attention to spare, I'll come back to this. As of now though, :shrug:

**Comment by Ammako** [comment]
Still a thing on 0.21.6 ;(

Hard to get logs given how intermittent the issues are, by the time I close the app, open terminal to stsrt logcat, and reopen the app, on second try it doesn't happen again, so my logcat doesn't actually get anything...

**Comment by Thewisem** [comment]
So good for you. I guess

**Comment by int3rw3bs** [comment]
> Can you guys test out this apk.
> It is a bare bones attempt to go with the solution from the youtube-dl thread that @conkerts linked here [#6510 (comment)](https://github.com/TeamNewPipe/NewPipe/issues/6510#issuecomment-871468397)
> I'll be using this version too for a few days. If it works, then I'll make a proper PR out of it.
> 
> Only videos are adjusted, audio streams are not adjusted
> 
> [app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/6793074/app-debug.zip)
> 
> Changes are:
> [XiangRongLin/NewPipeExtractor@8c11d23](https://github.com/XiangRongLin/NewPipeExtractor/commit/8c11d23fccb9a386193f4548e24b71a902161771)
> 
> EDIT: nope does not work

I have tested it out for a few days, it seems better, but eventually buffers in between videos, but at least with this version the buffering  period is short. Not perfect but an improvement.

**Comment by int3rw3bs** [comment]
> > @SeongGino Can you share the **streaming** URL and also the YouTube URL of a video on which you have the buffering issue? I want to check something.
> 
> The problem is, it's not just _a_ video that has the issue. It's _**any**_ video after the cache builds up over the course of a small handful of video streams. So I don't see how this would benefit, when/if you yourself could simply check any random video after using any build for a period.

Yes, this is exactly what happens. seems like it runs out of ram or hdd space to cache to me.

**Comment by AbsurdlySuspicious** [comment]
@SeongGino 

> could be gathered from any front page video without my intervention

Nope. Discussed apk contains some massive changes to extractor, that is, the thing that extracts needed info from the page and constructs a streaming url. This is not a trivial process in itself, and youtube page changes frequently and is never same for different users. If you don't experience any improvement, it means you've hit some corer case and streaming url from *your* device is needed to investigate the issue. Don't think you were asked to do this just for fun, there is a reason

**Comment by TheGrigoryans** [comment]
when will this problem be fixed? Already 2 months have passed, they got bored ... Why update 21.6 if the main thing is not solved?

**Comment by AndhikaWB** [comment]
Here's [my log file](https://pastebin.com/bRtHfR97) captured using [Logcat Reader](https://play.google.com/store/apps/details?id=com.dp.logcatapp). Unfiltered cuz I don't know what to filter, but I tried keeping it as small as I can.

**Some details**
Video URL: https://youtu.be/eb6ja96y6Ps
Android version: 10 (Resurrection Remix ARM)
NewPipe version: 0.21.6

Video started to buffer around 0:15 during this test (log only captured on a test). I played several other random videos as well, restarted NewPipe, and play the same set of videos again in random order to check if it's video specific. Testing 4 times, resulted only 2 times buffering issue [on this video](https://youtu.be/eb6ja96y6Ps) (but no buffering on other videos) so I can somewhat says it happened randomly.

**Comment by SmoothJaz** [comment]
It felt like years passed waiting for 21.6 and I think I piece of my soul left me once I got the buffering bug after updating :(

I'm not really sure how to help troubleshoot with information that hasn't already been said but man, this bug stinks big time.

For what it's worth:
Ver. 0.21.6
Samsung note 20 ultra
Android 11 One UI 3.1
(Can provide other specs I just don't know what's relevant here)

**Comment by njmdietrich** [comment]
So I don't know if this is still needed, but I thought I would share my logcat of this issue as well. The video in question was this: https://www.youtube.com/watch?v=Ss8_OGlOtXQ and it immediately started to occur when playing the video, so I paused, started the logcat recorder and resumed the video.
My device: Poco F1
Android 11 (lineageos for microg)
Newpipe 0.21.6

Here is the log:
`<<< log_count = 24 >>>
[07-16 13:39:18.187 2034:2056 W/Choreographer]
Frame time is 0.128807 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.

[07-16 13:38:45.165 0:0 I/FG]
calculate_average_current: current_now_ma=288574 averaged_iavg_ma=287353

[07-16 13:38:45.165 0:0 I/FG]
fg_battery_soc_smooth_tracking: soc:87, last_soc:87, raw_soc:87, soc_changed:1.

[07-16 13:38:45.165 0:0 I/FG]
soc_monitor_work: soc:87, raw_soc:87, c:288574, s:2

[07-16 13:39:18.942 2478:2478 V/DeadZone]
consuming errant click: (589.0,44.0)

[07-16 13:39:19.057 2967:2967 D/RecentsOrientedState]
current RecentsOrientedState: [this=RecentsOrientedState@d980c51 mOrientationHandler=PortraitPagedViewHandler@5c8c442 mDisplayRotation=0 mTouchRotation=0 mRecentsActivityRotation=0 isRecentsActivityRotationAllowed=false mSystemRotation=false mFlags=3]

[07-16 13:39:19.057 2967:2967 I/chatty]
uid=10110(com.android.launcher3) identical 2 lines

[07-16 13:39:19.057 2967:2967 D/RecentsOrientedState]
current RecentsOrientedState: [this=RecentsOrientedState@d980c51 mOrientationHandler=PortraitPagedViewHandler@5c8c442 mDisplayRotation=0 mTouchRotation=0 mRecentsActivityRotation=0 isRecentsActivityRotationAllowed=false mSystemRotation=false mFlags=3]

[07-16 13:39:19.058 2967:2967 D/RecentsOrientedState]
current RecentsOrientedState: [this=RecentsOrientedState@d976cb7 mOrientationHandler=PortraitPagedViewHandler@5c8c442 mDisplayRotation=0 mTouchRotation=0 mRecentsActivityRotation=0 isRecentsActivityRotationAllowed=false mSystemRotation=false mFlags=3]

[07-16 13:39:18.942 2478:2478 V/DeadZone]
consuming errant click: (589.0,44.0)

[07-16 13:39:19.062 2034:2052 D/AutofillManagerService]
Close system dialogs

[07-16 13:39:19.059 2967:2967 I/chatty]
uid=10110(com.android.launcher3) identical 2 lines

[07-16 13:39:19.060 2967:2967 D/RecentsOrientedState]
current RecentsOrientedState: [this=RecentsOrientedState@d976cb7 mOrientationHandler=PortraitPagedViewHandler@5c8c442 mDisplayRotation=0 mTouchRotation=0 mRecentsActivityRotation=0 isRecentsActivityRotationAllowed=false mSystemRotation=false mFlags=3]

[07-16 13:39:19.063 2478:2478 V/ShadeControllerImpl]
NotificationShadeWindow: com.android.systemui.statusbar.phone.NotificationShadeWindowView{12a906b I.E...... ......ID 0,0-1080,2246 aid=5} canPanelBeCollapsed(): false

[07-16 13:39:19.077 2967:2967 D/RecentsOrientedState]
current RecentsOrientedState: [this=RecentsOrientedState@d980c51 mOrientationHandler=PortraitPagedViewHandler@5c8c442 mDisplayRotation=0 mTouchRotation=0 mRecentsActivityRotation=0 isRecentsActivityRotationAllowed=false mSystemRotation=false mFlags=3]

[07-16 13:39:19.077 2967:2967 D/RecentsOrientedState]
current RecentsOrientedState: [this=RecentsOrientedState@d980c51 mOrientationHandler=PortraitPagedViewHandler@5c8c442 mDisplayRotation=0 mTouchRotation=0 mRecentsActivityRotation=0 isRecentsActivityRotationAllowed=false mSystemRotation=false mFlags=3]

[07-16 13:39:19.077 2967:2967 D/RecentsOrientedState]
current RecentsOrientedState: [this=RecentsOrientedState@d976cb7 mOrientationHandler=PortraitPagedViewHandler@5c8c442 mDisplayRotation=0 mTouchRotation=0 mRecentsActivityRotation=0 isRecentsActivityRotationAllowed=false mSystemRotation=false mFlags=3]

[07-16 13:39:19.077 2967:2967 D/RecentsOrientedState]
current RecentsOrientedState: [this=RecentsOrientedState@d976cb7 mOrientationHandler=PortraitPagedViewHandler@5c8c442 mDisplayRotation=0 mTouchRotation=0 mRecentsActivityRotation=0 isRecentsActivityRotationAllowed=false mSystemRotation=false mFlags=3]

[07-16 13:39:19.255 2967:2967 W/Choreographer]
Frame time is 0.301465 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.

[07-16 13:39:19.348 2967:2994 I/droid.launcher]
NativeAlloc concurrent copying GC freed 510054(13MB) AllocSpace objects, 0(0B) LOS objects, 62% free, 57MB/153MB, paused 53us total 244.669ms

[07-16 13:39:19.358 2967:2997 W/System]
A resource failed to call release. 

[07-16 13:39:19.359 2967:2997 I/chatty]
uid=10110(com.android.launcher3) FinalizerDaemon identical 4 lines

[07-16 13:39:19.359 2967:2997 W/System]
A resource failed to call release. 

[07-16 13:39:19.472 2967:2967 W/Choreographer]
Frame time is 0.329987 ms in the future!  Check that graphics HAL is generating vsync timestamps using the correct timebase.`

**Comment by njmdietrich** [comment]
I have had this issue for months now so I thought I would chime in and share a logcat as well. No idea what to filter for, so you are just getting all of it unfortunately... I noticed the issue occurring on a video, so I started the logcat record, went back to newpipe and resumed the video.
Link to the video in question: https://www.youtube.com/watch?v=Ss8_OGlOtXQ
My phone: Poco F1, Android 11 (LineageOS for MicroG)
NewPipe version: 0.21.6
logcat: https://pastebin.com/v8hLtniK

**Comment by bioblatt** [comment]
> I have had this issue for months now so I thought I would chime in and share a logcat as well. No idea what to filter for, so you are just getting all of it unfortunately... I noticed the issue occurring on a video, so I started the logcat record, went back to newpipe and resumed the video. There was buffering for a few seconds when resuming the video, and then it would just play half a second, buffer, rinse and repeat.
> Link to the video in question: https://www.youtube.com/watch?v=Ss8_OGlOtXQ
> My phone: Poco F1, Android 11 (LineageOS for MicroG)
> NewPipe version: 0.21.6
> logcat: https://pastebin.com/v8hLtniK

I have the exact same issue. Videos starts playing after 10-15 seconds and stops right away, plays and stops again after half a second.

**Comment by XiangRongLin** [comment]
Can you guys test out the apk inside #6673 for a few days and report back in that PR. I will not be responding to an y comments here.

**Comment by ouzowtf** [comment]
Seems Like yt-dlp and pytube have solved this problem, see this ytdl-thread and later comments: https://github.com/ytdl-org/youtube-dl/issues/29326#issuecomment-879256177

**Comment by LogoiLab** [comment]
This issue is occuring more often with the most recent update (0.21.6). Every video I load buffers, plays for a second, buffers, etc etc for almost the entire video. Some videos are just fine. But most do this buffering thing now. It was fine in 0.21.5 but just go so much worse in 0.21.6. 

I am on a samsung galaxy S9 Android 10. Connected via a 1Gbps DOCSIS3.1 link.

Things I've tried:

- Clear caches: no effect
- VPN: no effect
- Switch from MP4 to WebM: no effect
- Downgrade quality to 480p: small to no effect.

**Comment by noko00** [comment]
If buffering start. Go to comments. Then scroll comments until comments need  to download new lines. Back to video, no more buffering.

**Comment by noko00** [comment]
Good luck with Newpipe. I like Newpipe wrey much.

**Comment by infinitewaveparticle** [comment]
I, too, am experiencing buffering on all my devices with version 0.21.6 .After first opening the app the first 5-10 videos play fine. After that I can only watch smoothly at 240p or less which is like watching TV in the 80s. 360p and higher just buffers every 1-2s. This problem has been going on for about 2-3 months. Every once in a while, seemingly randomly, I can play a video in up to 2160p60 just fine. The next video it's back to 240p or even 144p sometimes. Like everyone else, this is the only app that does this. My devices are: NVIDIA Shield running stock Android 9 (TV), Samsung Galaxy S7 Edge running stock Android 8, and Kindle Fire HD 10 7th Edition tablet running stock Android 5.1.1. Internet connection FIOS Gigabit. I'll try the listed beta app above, but this issue is far from resolved in the 0.21.6 release version. The issue is exactly the same as in 0.21.5: first 5-10 videos are great, then none work except the rare random occurrence of one video playing as it should.

**Comment by irkiosan** [comment]
Same issue here. I had this issue already with the previous release. 21.6 made the issue worse. Currently, I barely can watch any video. It buffers right away even at low resolution. It is so bad, there isn't even 1 second of continuous playback.

Pixel 4 XL, CalyxOS, NewPipe 21.6

**Comment by TobiGr** [comment]
Please stop commenting that you have the same issue. That does not help at all. It just causes a bunch of people getting notifications and spending time on reading meaningless comments.
As already said, test the APK from #6673 and leave a comment / vote **there** telling us whether the problem is solved.

**Comment by Isssuemaker** [comment]
I have same issue but when I Clear Cache the problem is solved so try to clear cache

**Comment by pythonInRelay** [comment]
Newpipe has had this problem for well over a year. I suspect it is a cache issue. Clearing the cache always fixes the problem (at least for me). `Settings > History & Cache > (Wipe everything)`

Generally Newpipe bogs down for me after 50-odd videos watched. This is on a phone with 8GB of RAM and 256GB of storage. So neither of those matters.

As I said, wiping the cache solves the problem. I've got a gigabit connection and videos immediately start buffering at 25MB/s or more which is about the same on the official app.

Using an external player does NOT solve the problem for me. Neither does using an older ExoPlayer. The problem always comes back after a while.

I'm using a Xiaomi Mi 10T Pro and the same problem happens on my partner's Nokia 7.2.

**Comment by infinitewaveparticle** [comment]
As I said I plan to test 6673 apk, but first I need to fix a completely different issue preventing me from installing apps. That's why I haven't been able to yet.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by BoldyPlays** [comment]
Thanks pythonInRelay. Clearing cache temporarily resolves the issue. Galaxy note 10+ (SM-N975F) Android 11 with Newpipe 0.21.7

**Comment by deadcombo** [comment]
The apk labeled as youtubei-api fixed the issue for me. Testing for over a week.

**Comment by ksapp-dell** [comment]
The above mentioned APK worked better, but didn't solve the issue completely. Also the 21.7 is still buffering - but not for all videos. The resolution doesn't matter in this case. 
May be the videos use different codec and some of the are buffering ?
I tried skytube and this app dies not buffer at all. Also the videos do start a lot quicker. May be this is a hint ?

**Comment by TiA4f8R** [comment]
Can you try this APK and check if the issue is fixed? It's based on https://github.com/TeamNewPipe/NewPipeExtractor/pull/683/commits/d13f531b6ff985b5ca51166e0c59ef3dc4a87b59.

[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/6887051/app-debug.zip)

Thank you in advance.

**Comment by bibin-v2020** [comment]
> Can you try this APK and check if the issue is fixed? It's based on [TeamNewPipe/NewPipeExtractor@d13f531](https://github.com/TeamNewPipe/NewPipeExtractor/commit/d13f531b6ff985b5ca51166e0c59ef3dc4a87b59) and on the 0.21.6 version (the only change I made in the app is the extractor version).
> 
> [app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/6887051/app-debug.zip)
> 
> Thank you in advance.

Seems issue fixed on this apk! I've tested it for some hour's, till now no buffering issue reported

**Comment by XiangRongLin** [comment]
> 
> 
> Can you guys test out the apk inside #6673 for a few days and report back in that PR. I will not be responding to an y comments here.

This got finalized with some help of the other members. See the apk in the new PR #6792

**Comment by BedHedd** [comment]
> If buffering start. Go to comments. Then scroll comments until comments need to download new lines. Back to video, no more buffering.

I found this solution to work on Newpipe 0.21.7. 
Realme XT - Android 10

Even if I get #6744, clicking the retry seems to work. I get the buffering issues more for longer videos. My gut is saying it's a placebo, so I'll update if this continues to work.
Examples
https://www.youtube.com/watch?v=pkuxIy3kFZM
https://www.youtube.com/watch?v=eoAzGg-Si2Q

**Comment by ksapp-dell** [comment]
Hi, i tried both above links with 21.7 and there is no buffering (also with 1080 - no problem) - only the app doesn't load comments. Just an error and two buttons (reload and report).

**Comment by swarupbhanjachowdhury** [comment]
Same issue, redmi note 10. Android 11.

**Comment by xdIceBear** [comment]
Buffers like crazy even on a 1000Mbps connection. pretty annoying please guys look in to this. this is on the latest version in 480p

**Comment by Disqu3-mirOir-qt** [comment]
when fixed problem persit between some version , critical bug because apps unusable now ,

**Comment by ksapp-dell** [comment]
It was getting better with disabled comments for many videos, but this one keeps buffering without comments - doesn't matter which resolution.
https://www.youtube.com/watch?v=tw9VF1V-Lzc&t=148

**Comment by infinitewaveparticle** [comment]
@ksapp-dell I tried this too and after a while it just went back to buffering for most videos. Some didn't even load even at 144p. I've been testing the test apk (#6792) and so far so good.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by Dastardly-Entrench** [comment]
Now that the pr is merged, will there be a hotfix release?

**Comment by AbsurdlySuspicious** [comment]
Comments fix is still unmerged AFAIK

**Comment by Somadima** [comment]
Everything is fine until the Newpipe reaches ~70 MB of cache then there is a problem with buffering, a small observation.

**Comment by 20manas** [comment]
This issue is still present in version 0.21.8 of NewPipe. This should be reopened.

**Comment by infinitewaveparticle** [comment]
I, too, have noticed some videos don't ever load... But so far my fix has been to switch to a very low resolution (240 or 144) and then once it begins playing switch back to the high resolution. So far that's worked every time.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by Ammako** [comment]
For me I just have to clear cache and then it works.

I don't think any of this is helpful though, this is not new information.

**Comment by conkerts** [comment]
> 
> 
> Everything is fine until the Newpipe reaches ~70 MB of cache then there is a problem with buffering, a small observation.

My buffering & 70kb/s speed issues seem to be solved with the latest update.

Though my NP is currently at 128MB cache and is still fine. 
But occasionally nothing works and I have to force close the app and clear the cache to make any video load again.
I think the cache thing/problem is something different, but I agree, it is also extremely annoying.

**Comment by aleksfadini** [comment]
> This issue is still present in version 0.21.8 of NewPipe. This should be reopened.

Agreed!

**Comment by bibin-v2020** [comment]
Buffering issue is fixed for me after 0.21.8 update but sometimes videos don't even play, it keeps on loading, that could be different issue.

**Comment by infinitewaveparticle** [comment]
I was watching a 2.5hr music concert today using 0.21.8... Not only did it not load until I changed the bitrate, but it also stopped playing and started buffering about once every 4-6 minutes. Changing the rate - up or down - always fixed it, but the problem kept happening over and over again. So yes, this seems to be a different, yet possibly related issue.

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by opusforlife2** [comment]
The problem this issue was opened for has been fixed in the current release. If you face any new problem, please investigate it thoroughly from scratch, assuming it's a different problem than this, and open a new issue if needed.
